src/Orleans/GrainDirectory/IGrainDirectory.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.GrainDirectory
+{
+
+    /// <summary>
+    /// Recursive distributed operations on grain directories.
+    /// Each operation may forward the request to a remote owner, increasing the hopCount.
+    /// 
+    /// The methods here can be called remotely (where extended by IRemoteGrainDirectory) or
+    /// locally (where extended by ILocalGrainDirectory)
+    /// </summary>
+    interface IGrainDirectory
+    {
+        /// <summary>
+        /// Record a new grain activation by adding it to the directory.
+        /// <para>This method must be called from a scheduler thread.</para>
+        /// </summary>
+        /// <param name="address">The address of the new activation.</param>
+        /// <param name="singleActivation">If true, use single-activation registration</param>
+        /// <param name="hopCount">Counts recursion depth across silos</param>
+        /// <returns>The registered address and the version associated with this directory mapping.</returns>
+        Task<AddressAndTag> RegisterAsync(ActivationAddress address, bool singleActivation, int hopCount = 0);
```

Comment:
Yes, this function combines the other two into a single body (to avoid code duplication). The intention is that the behavior remains the same as before... the parameter singleActivation switches between the two versions.

I may have forgotten to update the xml comment accordingly.